sycl/include/sycl/detail/vector_arith.hpp

```diff
@@ -228,25 +218,26 @@ class vec_arith : public vec_arith_common<DataT, NumElements> {
   friend std::enable_if_t<(COND), vec<ocl_t, NumElements>> operator RELLOGOP(  \
       const vec_t & Lhs, const vec_t & Rhs) {                                  \
     vec<ocl_t, NumElements> Ret{};                                             \
-    /* This special case is needed since there are no standard operator||   */ \
-    /* or operator&& functions for std::array.                              */ \
-    if constexpr (vec_t::IsUsingArrayOnDevice &&                               \
-                  (std::string_view(#RELLOGOP) == "||" ||                      \
-                   std::string_view(#RELLOGOP) == "&&")) {                     \
+    /* ext_vector_type does not support bfloat16, so for these   */            \
+    /* we do element-by-element operation on the underlying std::array.  */    \
+    if constexpr (vec_t::IsBfloat16) {                                         \
       for (size_t I = 0; I < NumElements; ++I) {                               \
         /* We cannot use SetValue here as the operator is not a friend of*/    \
         /* Ret on Windows. */                                                  \
-        Ret[I] = static_cast<ocl_t>(                                           \
-            -(vec_data<DataT>::get(detail::VecAccess<vec_t>::getValue(Lhs, I)) \
-                  RELLOGOP vec_data<DataT>::get(                               \
-                      detail::VecAccess<vec_t>::getValue(Rhs, I))));           \
+        Ret[I] = static_cast<ocl_t>(-(Lhs[I] RELLOGOP Rhs[I]));                \
```

Comment:
Is the comment on line 225 still accurate?

Reply:
Removed in https://github.com/intel/llvm/pull/14130/commits/ee589312ff65c2b3e95ae6c3e50cf2b5cc0612b5